core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -1949,12 +1812,12 @@ public synchronized FileSystemCommand workerHeartbeat(long workerId, List<Long>
     }
 
     // get the files for the given worker to checkpoint
-    List<PersistFile> filesToCheckpoint = pollFilesToCheckpoint(workerId);
-    if (!filesToCheckpoint.isEmpty()) {
-      LOG.debug("Sent files {} to worker {} to persist", filesToCheckpoint, workerId);
+    List<PersistFile> filesToPersist = mAsyncPersistHandler.pollFilesToPersist(workerId);
```

Comment:
I'm curious. Why is this workerHeartbeat() method the only synchronized method? Everything else synchronizes on the inode tree. Could this be a bug?